firmware/application/apps/ui_settings.cpp

```diff
@@ -726,6 +730,39 @@ void SetConfigModeView::focus() {
     button_save.focus();
 }
 
+/* FakeBrightnessView ************************************/
+
+SetFakeBrightnessView::SetFakeBrightnessView(NavigationView& nav) {
+    add_children({&labels,
+                  &field_fake_brightness,
+                  &button_save,
+                  &button_cancel,
+                  &checkbox_brightness_switch});
+
+    field_fake_brightness.set_by_value(pmem::fake_brightness_level());
+    checkbox_brightness_switch.set_value(pmem::apply_fake_brightness());
+
+    checkbox_brightness_switch.on_select = [this](Checkbox&, bool v) {
+        if (!v) {  // this is to hide status bar sun icon when check off, but don't de-hide it when check on (some user may don't wan't the icon, so this makes mor sense then !v)
+            pmem::set_ui_hide_fake_brightness(true);
```

Comment:
Sorry, one more thing I didn't notice earlier:  Personally I don't think this function should automatically hide or un-hide the status bar icon.  I would leave hiding or un-hiding to the user, since they already have this capability in the User Interface Settings.

Reply:
after thinking, i think this is good idea. fixed. 
thx!